### ISSUE TITLE
[Bug]: Opening a bookmark in private browsing mode doesn't work if there are no existing private tabs

### ISSUE THREAD

**Issue by delxg** [mozilla-mobile/fenix#23746]
### Steps to reproduce

1. Close all tabs in private browsing mode
2. Press and hold the tab switcher button and select "New private tab"
3. Dismiss the keyboard popup
4. Press the menu button, select "Bookmarks" and select a bookmark

### Expected behaviour

The bookmark should open in a new private tab

### Actual behaviour

Firefox switches out of private browsing mode and shows the previously opened tab. The bookmark is not loaded at all.

### Device name

Nokia X20

### Android version

Android 11

### Firefox release type

Firefox Nightly

### Firefox version

99.0a1 (Build #2015863467)

### Additional information

This doesn't happen in the current Firefox release version 97.1.0. I believe this is caused by #23169.

Thanks :-)

**Comment by LaurentiuApahideanSV** [comment]
I can confirm the issue is reproducible on Firefox Preview Nightly 2022-02-16.

### Devices used:
- Google Pixel 6 (Android 12)
- OnePlus 6T (Android)

**Comment by gtlittlewing** [comment]
PR #23169("For #13336: Open bookmarks in current tab") shows that the code was merged at 2022-02-03 4:58. So, I went to the [Nightly apk file repository](https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.nightly.latest), 
I installed the apk files built between 2022-02-02 and 2022-02-04 one by one from there, and tried to reproduce this issue.

|Build|Result|apk file|
|-----|------|---------------|
|98.0a1 (Build #2015860971), d6cfe0cde+ 2022-02-02T05:13:13.226095|Not reproduced|[target.apk](https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.2022.02.02.revision.d6cfe0cde04e47833ed5c16fa70013e8e305b7be.arm64-v8a/artifacts/public%2Fbuild%2Farm64-v8a%2Ftarget.apk )|
|98.0a1 (Build #2015861075), 4780ebaf5+ 2022-02-02T18:49:05.074691|Not reproduced|[target.apk](https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.2022.02.02.revision.4780ebaf5024896e15abf28ffe692068af9ae964.arm64-v8a/artifacts/public%2Fbuild%2Farm64-v8a%2Ftarget.apk)|
|98.0a1 (Build #2015861163), e73deb23a+ 2022-02-03T05:08:04.489098|`Reproduced`|[target.apk](https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.2022.02.03.revision.e73deb23ace1ecbe36e476de811de221129fc60f.arm64-v8a/artifacts/public%2Fbuild%2Farm64-v8a%2Ftarget.apk)|
|98.0a1 (Build #2015861259), 0be470a10+ 2022-02-03T17:10:35.031753|`Reproduced`|[target.apk](https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.2022.02.03.revision.0be470a104eca5a7cc450244b83e6ac2a6b12a04.arm64-v8a/artifacts/public%2Fbuild%2Farm64-v8a%2Ftarget.apk)|
|98.0a1 (Build #2015861451), e290bc1bd+ 2022-02-04T17:13:22.622130|`Reproduced`|[target.apk](https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.2022.02.04.revision.e290bc1bd6c4a9e3185e8f0bf9ffb433d378620d.arm64-v8a/artifacts/public%2Fbuild%2Farm64-v8a%2Ftarget.apk)|
|98.0a1 (Build #2015861467), de6e2e763+ 2022-02-04T19:46:04.287038|`Reproduced`|[target.apk](https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.2022.02.04.revision.de6e2e763e756b4a9b775375f4c985d83c7fc08e.arm64-v8a/artifacts/public%2Fbuild%2Farm64-v8a%2Ftarget.apk)|

### Device Used:

- SO-52A (NTT DoCoMo variant of XPERIA 5 II,Android 11)

I hope the above is helpful.

**Comment by czlucius** [comment]
I think the problems caused by opening bookmarks in the current tab is greater than the problems caused by opening them in a new tab, currently.
@jonalmeida sorry for mentioning you several times, but should this be looked into again? 
If it's possible, the behaviour can be determined by whether the user is on the home screen. I've tried checking for the currently visible fragment as per https://stackoverflow.com/questions/16652095/how-to-know-if-a-fragment-is-visible/16652217#16652217 something like this(not the actual code):
```kotlin 
val isUserOnHomeScreen = activity.supportFragmentManager.findFragmentById(R.id.homeFragment)?.isVisible() 
```
without any success. Do you know of a way where this is possible (checking if the user is on the Home Fragment)?

Thanks

**Comment by jonalmeida** [comment]
@czlucius the fragment does not know the state of the mode you're in. The BrowsingManager, that you pass into the `openToBrowserAndLoad` function.

A quick inspection into that code, tells me that the `newTab` flag logic doesn't consider that the browsing mode has changed, so it loads into the tab of the currently selected session:

https://github.com/mozilla-mobile/fenix/blob/d599cc7375a00ef808628cd36bc27c8a0db66152/app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt#L155-L170

What you would need to do is probably update that logic to make it respect the current mode.

**Comment by SoftVision-LorandJanos** [comment]
I could reproduce this issue on the Nightly 99.0a1 from 3 days ago (2022-02-21).
Verified as fixed on the latest Nightly 99.0a1 (2022-02-24).
Device used: Oppo Reno 6 (Android 11).
Closing the ticket as fixed.
